packages/microsoft-security-adapter/src/filters/entra/application_standalone_fields.ts

```diff
@@ -0,0 +1,504 @@
+    standaloneFieldsInstances: Object.values(clonedParentToStandaloneFields).flat(),
+    standaloneFieldsChanges,
+  })
+
+  const fullNamesToExistingParentChanges = Object.fromEntries(
+    existingParentChanges.map(change => [getChangeData(change).elemID.getFullName(), change]),
+  )
+  const parentChangesToDeploy = parentsWithStandaloneFieldsChanges
+    .map(parent => {
+      const parentChange = fullNamesToExistingParentChanges[parent.elemID.getFullName()] ?? {
+        action: 'modify' as const,
+        data: {
+          before: parent,
+          after: parent.clone(),
+        },
+      }
+      if (isRemovalChange(parentChange)) {
+        // No need to update the parent change with the standalone fields, since it's a removal change, and the standalone fields will be removed as well.
+        return parentChange
+      }
+
+      updateParentChangeWithStandaloneFields({
+        parentChange,
+        standaloneFields: clonedParentToStandaloneFields[parent.elemID.getFullName()] ?? [],
+      })
+
+      return parentChange
+    })
+    .filter(isDefined)
+
+  return parentChangesToDeploy
+}
+
+// Calculate the deploy result:
+// The deploy request only includes the parent changes, so we need to add the standalone fields changes to the result, if their parent change was applied.
+// We also need to filter applied parent changes that didn't originally have changes.
+const calculateDeployResult = ({
+  deployResult: { errors, appliedChanges: appliedParentChanges },
+  standaloneFieldsChanges,
+  existingParentChanges,
+  instancesWithNoParentErrors,
+}: {
+  deployResult: DeployResult
+  standaloneFieldsChanges: Change<InstanceElement>[]
+  existingParentChanges: Change<InstanceElement>[]
+  instancesWithNoParentErrors: SaltoElementError[]
+}): DeployResult => {
+  const existingParentChangesFullNames = new Set(
+    existingParentChanges.map(change => getChangeData(change).elemID.getFullName()),
+  )
+  const appliedExistingParentChanges = appliedParentChanges.filter(change =>
+    existingParentChangesFullNames.has(getChangeData(change).elemID.getFullName()),
+  )
+
+  const appliedParentChangesFullNames = new Set(
+    appliedParentChanges.map(change => getChangeData(change).elemID.getFullName()),
+  )
+  const appliedStandaloneFieldsChanges = standaloneFieldsChanges.filter(change =>
+    appliedParentChangesFullNames.has(getParents(getChangeData(change))[0]?.elemID.getFullName()),
+  )
+
+  return {
+    errors: errors.concat(instancesWithNoParentErrors),
+    appliedChanges: appliedExistingParentChanges.concat(appliedStandaloneFieldsChanges),
+  }
+}
+
+/*
+ * Add parent_id to the newly added standalone fields.
+ * This is needed to ensure uniqueness in serviceId for those fields, whose IDs uniqueness is parent-contextual.
+ */
+const addParentIdToChanges = async (changes: Change[]): Promise<void> => {
+  const relevantChanges = changes.filter(isInstanceChangeOfRelevantTypes).filter(isAdditionOrModificationChange)
```

Comment:
should be only addition changes, no ?
```suggestion
  const relevantChanges = changes.filter(isInstanceChangeOfRelevantTypes).filter(isAdditionChange)
```